content/en/docs/ops/configuration/traffic-management/network-topologies/index.md

```diff
@@ -195,3 +174,33 @@ where `ENUM_VALUE` can be of the following type.
 
 See the [Envoy documentation](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_conn_man/headers#x-forwarded-client-cert)
 for examples of using this capability.
+
+## PROXY Protocol
+
+The [PROXY protocol](https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt) allows for exchanging and preservation of client attributes across multiple proxies without relying on Layer 7 protocols.
+
+Enabling this requires adding [Envoy Proxy Protocol filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/proxy_protocol), using an `EnvoyFilter` applied on the gateway workload. For example:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: proxy-protocol
+  namespace: istio-system
+spec:
+  configPatches:
+  - applyTo: LISTENER
+    patch:
+      operation: MERGE
+      value:
+        listener_filters:
+        - name: envoy.listener.proxy_protocol
+        - name: envoy.listener.tls_inspector
+  workloadSelector:
+    labels:
+      istio: ingressgateway
+{{< /text >}}
+
+{{< idea >}}
+Note that the PROXY protocol can be used in conjunction with the `gatewayTopology` configuration, and the `numTrustedProxies` will be used to determine the address for the `X-Envoy-External-Address` header.
```

Comment:
```suggestion
Note that the PROXY protocol can be used in conjunction with the `gatewayTopology` configuration, and the `numTrustedProxies` will be used to determine the address for the `X-Forwarded-For` header.
```

Reply:
My take away from Lin's tests:

When proxy protocol with gateway listener filter is set and
1. no XFF headers sent, no gatewayTopology = X-Envoy-External-Address not set, XFF set to client IP
2. with XFF headers, no gatewayTopology = X-Envoy-External-Address set, XFF set to client IP
3. with XFF headers, with gatewayTopology = X-Envoy-External-Address set to configured hops, XFF has client IP appended

Without proxy protocol, 
4. no XFF headers sent, no gatewayTopology = X-Envoy-External-Address not set, XFF set to gw IP
5. with XFF headers sent, no gatewayTopology = X-Envoy-External-Address not set, XFF has gw IP appended
6.  no XFF headers, with gatewayTopology = X-Envoy-External-Address not set, XFF set to gw IP
7. with XFF headers, with gatewayTopology = X-Envoy-External-Address set to configured hops, gw IP appended

(...a table might have been a better choice)

This Note was meant to talk about 3.... XFF has client IP appended, but XEEA is set based on numTrustedProxies. 

Agree? We can add another sentence in the description of Proxy Protocol that the XFF value will be set to the client address (or appended).

Reply:
Mostly. Minor tweaks:

When proxy protocol with gateway listener filter is set and

2. with XFF headers, no gatewayTopology = X-Envoy-External-Address set, XFF set to client IP -> with XFF headers, no gatewayTopology = X-Envoy-External-Address set, XFF has client IP appended